src/OrbitGl/TimeGraph.cpp

```diff
@@ -867,10 +867,19 @@ void TimeGraph::DoDraw(orbit_gl::PrimitiveAssembler& primitive_assembler,
                                         kGreenLineColor);
   }
 
-  // Right vertical margin of the Tracks to make them look nicer. If the vertical slider is visible,
-  // the margin will be between the slider and the tracks.
-  Vec2 right_margin_pos{GetWidth() - GetRightMargin(), GetPos()[1]};
+  DrawMarginsBetweenChildren(primitive_assembler);
+}
 
+void TimeGraph::DrawMarginsBetweenChildren(
```

Comment:
Why do we explicitly need to draw boxes into the margins?

Reply:
Really good question. Sorry, in the refactor I erased a bug that I shouldn't have erased. We didn't solve the issue of elements being drawn outside its area. Thanks for catching it.

I included the bug again and a better comment.

You can see for example very clearly the issue in FrameTracks if I erase the right margin box (http://screenshot/5nmXY5KYEwTNUeP)